### PR TITLE
REF: SingleBlockManager dont subclass BlockManager

### PR DESCRIPTION
**Pull request by jbrockmendel**
Instead, most of BlockManager becomes _BlockManager which is subclassed by SingleBlockManager and BlockManager, the latter holding only the 2d-specific BM methods.

Motivation: putting BlockManager constructor in cython gives a much bigger gain if we can define `__cinit__`, and we can't do that with the two subclasses having mismatched `__init__` signatures.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@jreback gentle ping, this is preliminary to some nice perf (constructor) improvements (along with #40776, #40757)

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -80,10 +81,10 @@
 
 # TODO: flexible with index=None and/or items=None
 
-T = TypeVar("T", bound="BlockManager")
+T = TypeVar("T", bound="_BlockManager")
 
 
-class BlockManager(DataManager):
+class _BlockManager(DataManager):
```
Comment (jreback) [inline pandas/core/internals/managers.py:87]: can you indicate that this is an abstract class (alt put ABC in the name)

File: pandas/core/internals/managers.py [hunk 2]
```diff
@@ -1474,21 +1387,123 @@ def take(self: T, indexer, axis: int = 1, verify: bool = True) -> T:
             consolidate=False,
         )
 
-    def _equal_values(self: T, other: T) -> bool:
+
+class BlockManager(_BlockManager):
```
Comment (jreback) [inline pandas/core/internals/managers.py:1391]: why don't we actually name it  Block2DManager then

File: pandas/core/internals/managers.py [hunk 3]
```diff
@@ -1474,21 +1387,123 @@ def take(self: T, indexer, axis: int = 1, verify: bool = True) -> T:
...
+                    f"number of axes ({self.ndim})"
+                )
+
+        if verify_integrity:
+            self._verify_integrity()
+
+        # Populate known_consolidate, blknos, and blklocs lazily
+        self._known_consolidated = False
+        # error: Incompatible types in assignment (expression has type "None",
+        # variable has type "ndarray")
```
Comment (jreback) [inline pandas/core/internals/managers.py:1420]: why do you need all these?

File: pandas/core/internals/managers.py [hunk 4]
```diff
@@ -1474,21 +1387,123 @@ def take(self: T, indexer, axis: int = 1, verify: bool = True) -> T:
...
+        self._blknos = None  # type: ignore[assignment]
+        # error: Incompatible types in assignment (expression has type "None",
+        # variable has type "ndarray")
+        self._blklocs = None  # type: ignore[assignment]
+
+    @classmethod
+    def _simple_new(cls, blocks: tuple[Block, ...], axes: list[Index]):
```
Comment (jreback) [inline pandas/core/internals/managers.py:1427]: shouldn't you have an abc for this?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/managers.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:1391]: bc (to my chagrin) downstream packages access it

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:87]: good idea

File: pandas/core/internals/managers.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:1427]: the attributes or the "type ignore"s?

File: pandas/core/internals/managers.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py:1434]: nothing in the base class uses _simple_new, so no

File: pandas/core/internals/managers.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/internals/managers.py]: updated + green

File: pandas/core/internals/managers.py (hunk 3 above)
Comment (jreback) [reply pandas/core/internals/managers.py:1427]: the type ignores, but certainly can be in a followup

**Review by jreback — APPROVED** [review]
couple of comments, but certailnly can be in followups

**Pull request merged**